pkg/api/proto/runme/notebook/v1alpha1/notebook.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package runme.notebook.v1alpha1;
+
+import "google/protobuf/wrappers.proto";
+import "runme/parser/v1/parser.proto";
+import "runme/runner/v1/runner.proto";
+
+option go_package = "github.com/stateful/runme/v3/pkg/api/gen/proto/go/runme/notebook/v1alpha1;notebookv1alpha1";
+
+message ResolveNotebookRequest {
+  runme.parser.v1.Notebook notebook = 1;
+
+  runme.runner.v1.CommandMode command_mode = 2;
+
+  google.protobuf.UInt32Value cell_index = 3;
+}
+
+message ResolveNotebookResponse {
+  string script = 1;
+}
+
+service NotebookService {
+  // ResolveNotebook that takes a notebook and a cellIndex and returns a script
+  // that takes the whole notebook into account to run as cell.
+  rpc ResolveNotebook(ResolveNotebookRequest) returns (ResolveNotebookResponse) {}
```

Comment:
Question: Is there any reason this method couldn't be in `RunnerService`?

Reply:
That's where I had it first. I decided to make it a separate service because the `Runner` only knows about atomic cells, and this service needs the entire notebook. Perhaps that's not "different enough". We could consider bringing it back into the Runner as part of runnerv2.